s4e-backend/src/main/java/pl/cyfronet/s4e/service/InstitutionService.java

```diff
@@ -129,6 +135,10 @@ public void addInstitutionAdminIfRequested(String adminMail, String institutionS
         return institutionRepository.findAllBy(projection);
     }
 
+    public <T> Optional<T> findBySlug(String slug, Class<T> projection) {
+        return institutionRepository.findBySlug(slug, projection);
+    }
```

Comment:
The same is done in ::getInstitution, however, I actually find ::findBySlug name better. I'd remove all ::getInstitution usages and replace them with ::findBySlug